server/src/main/java/org/elasticsearch/cluster/metadata/MetadataDataStreamsService.java

```diff
@@ -362,6 +407,21 @@ public static ClusterState setRolloverOnWrite(
         );
     }
 
+    public void updateSettings(
+        ProjectResolver projectResolver,
```

Comment:
Can we pass the project ID directly here instead of the project resolver? That avoids a dependency from this class on the resolver and it's in line with what we're doing in other places (both this class and other classes).